src/AppBundle/Controller/BackupDestinationController.php

```diff
@@ -0,0 +1,444 @@
+     *              type="string"
+     *          ),
+     *          @OAS\Property(
+     *              property="username",
+     *              type="string"
+     *          ),
+     *          @OAS\Property(
+     *              property="password",
+     *              type="string"
+     *          ),
+     *      ),
+     *  ),
+     * )
+     *
+     * @param Request $request
+     * @param integer $destId
+     * @param EntityManagerInterface $em
+     * @return Response
+     * @throws ElementNotFoundException
+     * @throws WrongInputException
+     */
+    public function updateBackupDestinationAction(Request $request, int $destId, EntityManagerInterface $em)
+    {
+        $dest = $this->getDoctrine()->getRepository(BackupDestination::class)->find($destId);
+
+        if (!$dest) {
+            throw new ElementNotFoundException(
+                'No backup destination found for id ' . $destId
+            );
+        }
+
+        $dest->setName($request->get('name'));
+        $dest->setDescription($request->get('description'));
+        $dest->setHostname($request->get('hostname'));
+        $dest->setProtocol($request->get('protocol'));
+        $dest->setPath($request->get('path'));
+
+        if ($request->request->has("username")) {
+            $dest->setUsername($request->get('username'));
+        }
+        if ($request->request->has("password")) {
+            $dest->setPassword($request->get('password'));
+        }
+
+        $this->validation($dest);
+
+        $em->flush();
+
+        $serializer = $this->get('jms_serializer');
+        $response = $serializer->serialize($dest, 'json');
+        return new Response($response);
+    }
+
+
+    /**
+     * Delete one backup destination
+     *
+     * @Route("/backupdestinations/{destId}", name="backup_dest_delete", methods="DELETE")
+     *
+     * @OAS\Delete(path="/backupdestinations/{destId}",
+     *  tags={"backupdestinations"},
+     *  @OAS\Parameter(
+     *     description="ID der zu löschenden backup destination",
+     *     in="path",
+     *     name="destId",
+     *     required=true,
+     *     @OAS\Schema(
+     *         type="integer"
+     *     ),
+     *  ),
+     *
+     *  @OAS\Response(
+     *     response=204,
+     *     description="success message"
+     *  ),
+     *
+     *  @OAS\Response(
+     *      response=404,
+     *      description="No backup destination found",
+     *  ),
+     * )
+     *
+     * @param integer $destId
+     * @param EntityManagerInterface $em
+     * @return JsonResponse
+     * @throws ElementNotFoundException
+     */
+    public function deleteBackupDestinationAction(int $destId, EntityManagerInterface $em)
+    {
+        $dest = $this->getDoctrine()->getRepository(BackupDestination::class)->find($destId);
+
+        if (!$dest) {
+            throw new ElementNotFoundException(
+                'No backup destination found for id ' . $destId
+            );
+        }
+
```

Comment:
Muss nicht noch validiert werden, ob die BackupDestination noch in einem BackupSchedule genutzt wird?

Reply:
Soll der Vorgang bei Verknüpfung mit BackupSchedule abgebrochen werden?

Reply:
Wäre das beste